tictactoe/implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/implementation/Game.kt

```diff
@@ -0,0 +1,93 @@
+package com.anaplan.engineering.azuki.tictactoe.implementation
+
+import org.slf4j.LoggerFactory
+
+class Game(
+    val board: MutableList<MutableList<Token?>>,
+    val playOrder: PlayOrder,
+) {
+
+    val width by lazy { board[0].size }
+    val height by lazy { board.size }
+
+    private val lines by lazy {
+        mutableListOf<List<Pair<Int, Int>>>().apply {
+            addAll((0 until height).map { y -> (0 until width).map { x -> y to x } })
+            addAll((0 until width).map { x -> (0 until height).map { y -> y to x } })
+            if (width == height) {
+                add((0 until width).map { i -> i to i })
+                add((0 until width).map { i -> width - 1 - i to i })
+            }
+        }
+    }
+
+    private fun getWinningLine() =
+        lines.find { line ->
+            line.map { board[it.first][it.second] }.toSet().singleOrNull() != null
+        }
+
+    private val winner: Token?
+        get() {
+            val line = getWinningLine()
+            val arbitraryPositionOnLine = line?.first()
+            return if (arbitraryPositionOnLine == null) {
+                null
+            } else {
+                board[arbitraryPositionOnLine.first][arbitraryPositionOnLine.second]
+            }
+        }
+
+    override fun toString() =
+        board.joinToString("\n") { row ->
+            row.joinToString(" | ") { it?.symbol ?: "." }
+        }
+
+    fun move(player: Player, x: Int, y: Int) {
+        Log.info("$player moves to (row $y, col=$x)")
+        if (canMove(player, x, y)) {
+            throw IllegalArgumentException("Coordinates out of bounds (row $y, col=$x)")
```

Comment:
🚫 I think a negation is missing.
```suggestion
        if (!canMove(player, x, y)) {
            throw IllegalArgumentException("Coordinates out of bounds (row=$y, col=$x)")
```

Reply:
The error message could also be improved, since a player cannot make a move also if the target position is already occupied.

Reply:
Again, the intent is not for a complete impl, but to have tests passing for merge